### ISSUE TITLE
Error during amp deployment and source code compile

### ISSUE THREAD

**Issue by teqnology** [Alfresco/analytics-tools#1]
I just tried to install the amp on Alfresco Enterprise 4.2.2 and this is the error:    

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'randomEnricherHelper' defined in URL [jar:file:/deployments/pingdomalf/alfresco/tomcat/webapps/alfresco/WEB-INF/lib/analytics-tools.jar!/org/alfresco/analytics/event/RandomEnricherHelper.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Could not instantiate bean class [org.alfresco.analytics.event.RandomEnricherHelper]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: org/apache/commons/math3/distribution/IntegerDistribution
```

Also, it seems like the pom.xml is not correct either for using alfresco enterprise artifacts, and other artifacts that are missing (dependecies missing error during ./run.sh execute). Been trying to use it with enterprise artifacts for Alfresco 5 (Maven 2.0.0).

**Comment by covolution** [comment]
Alfresco Analytics runs on _5.0 Enteprise +_. The dependency is on commons-math which alfresco-core depends on.  I will update the readme.

**Comment by teqnology** [comment]
I made some progress, although the documentation seems not to be updated or complete.
mvn package fails as apparently now the -Penterprise value must be used with Maven SDK 2.0.0. Also, the alfresco-private-repository id is not valid anymore and alfresco-private (or alfresco-internal, fixed in 2.0.1) has to be configured in .m2/settings.xml file. So `mvn package -Penterprise`, as `mvn package` is throwing errors:

```
[ERROR] Failed to execute goal on project analytics-tools: Could not resolve dependencies for project org.alfresco.analytics:analytics-tools:amp:1.0.1-SNAPSHOT: The following artifacts could not be resolved: org.alfresco:alfresco-enterprise:jar:classes:5.0, org.alfresco:alfresco-repository:jar:tests:5.0, org.alfresco.services:alfresco-events-repo-publishing:jar:tests:1.2.4: Failure to find org.alfresco:alfresco-enterprise:jar:classes:5.0 in https://artifacts.alfresco.com/nexus/content/groups/public was cached in the local repository, resolution will not be reattempted until the update interval of alfresco-public has elapsed or updates are forced -> [Help 1]
```

I'm adding the second issue here:

`mvn integration-test -Pamp-to-war,enterprise`

produces this error:

```
 2015-06-16 12:33:34,993  ERROR [domain.schema.SchemaBootstrap] [localhost-startStop-1] Unable to validate database schema.
```

 java.lang.IllegalArgumentException: dbObject cannot be null.
    at org.alfresco.util.schemacomp.DbProperty.<init>(DbProperty.java:51)

so apparently Alfresco 5 Enterprise still fails during project build. 

Is there anything I'm missing here?
I tried to update the pom.xml as stated here:

`https://issues.alfresco.com/jira/browse/MNT-13406`

but no luck.

Env:
Enterprise 5.0
Linux x64 14.04
Maven SDK 2.0.0

**Comment by covolution** [comment]
You appear to be having trouble building Alfresco from source using the Maven SDK.  This project "analytics-tools" takes an existing enterprise installation and applies an amp, it does not build Alfresco from source.  If you have build/SDK issues then https://github.com/Alfresco/alfresco-sdk/issues is a more appropriate project.

Once you have Alfresco 5.0 artifacts you can apply the amp.
